docs/conf.py

```diff
@@ -0,0 +1,275 @@
+
+# -- Options for intersphinx extension ---------------------------------------
+
+intersphinx_mapping = {
+    'python': ('https://docs.python.org/3', None),
+    'python2': ('https://docs.python.org/2', None),
+}
+
+# -- Options for todo extension ----------------------------------------------
+
+# If true, `todo` and `todoList` produce output, else they produce nothing.
+todo_include_todos = True
+
+# -- Options for sphinxcontrib.apidoc extension ------------------------------
+
+apidoc_excluded_paths = [
+    'plugin/cache/*',
+    'testing/*.py',
+]
+apidoc_extra_args = [
+    '--implicit-namespaces',
+    '--private',  # include “_private” modules
+]
+apidoc_module_dir = str(PROJECT_ROOT_DIR / 'proxy')
+apidoc_module_first = False
+apidoc_output_dir = 'pkg'
+apidoc_separate_modules = True
+apidoc_toc_file = None
+
+# -- Options for sphinxcontrib.spelling extension ----------------------------
+
+spelling_ignore_acronyms = True
+spelling_ignore_importable_modules = True
+spelling_ignore_pypi_package_names = True
+spelling_ignore_python_builtins = True
+spelling_ignore_wiki_words = True
+spelling_show_suggestions = True
+spelling_word_list_filename = [
+    'spelling_wordlist.txt',
+]
+
+# -- Options for extlinks extension ------------------------------------------
+
+extlinks = {
+    'issue': (f'{github_repo_url}/issues/%s', '#'),  # noqa: WPS323
+    'pr': (f'{github_repo_url}/pull/%s', 'PR #'),  # noqa: WPS323
+    'commit': (f'{github_repo_url}/commit/%s', ''),  # noqa: WPS323
+    'gh': (f'{github_url}/%s', 'GitHub: '),  # noqa: WPS323
+    'user': (f'{github_sponsors_url}/%s', '@'),  # noqa: WPS323
+}
+
+# -- Options for linkcheck builder -------------------------------------------
+
+linkcheck_ignore = [
+    r'http://localhost:\d+/',  # local URLs
+]
+linkcheck_workers = 25
+
+# -- Options for myst_parser extension ------------------------------------------
+
+myst_enable_extensions = [
+    'colon_fence',  # allow to optionally use ::: instead of ```
+    'deflist',
+    'html_admonition',  # allow having HTML admonitions
+    'html_image',  # allow HTML <img> in Markdown
+    # FIXME: `linkify` turns "Proxy.Py` into a link so it's disabled now
+    # Ref: https://github.com/executablebooks/MyST-Parser/issues/428#issuecomment-970277208
+    # "linkify",  # auto-detect URLs @ plain text, needs myst-parser[linkify]
+    'replacements',  # allows Jinja2-style replacements
+    'smartquotes',  # use "cursive" quotes
+    'substitution',  # replace common ASCII shortcuts into their symbols
+]
+myst_substitutions = {
+  'project': project,
+}
+
+# -- Strict mode -------------------------------------------------------------
+
+# The reST default role (used for this markup: `text`) to use for all
+# documents.
+# Ref: python-attrs/attrs#571
+default_role = 'any'
+
+nitpicky = True
+_any_role = 'any'
+_py_class_role = 'py:class'
+nitpick_ignore = [
+    (_any_role, '<proxy.HttpProxyBasePlugin>'),
```

Comment:
@abhinavsingh these are weird things mostly in single backticks found across the codebase. They all should be fixed one way or another, this is why they are ignored for now. But the action items are the same as for those linting violations for flake8.